stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -246,23 +247,28 @@ internal constructor(
                 else -> Unit // Ignore other events
             }
 
-            val currentUser = when {
-                event is HasOwnUser -> event.me
-                event is UserEvent && event.user.id == (getCurrentUser()?.id ?: "") -> event.user
-                else -> null
-            }
-            currentUser?.let { updatedCurrentUser ->
-                userStateService.onUserUpdated(updatedCurrentUser)
+            event.extractCurrentUser()?.let { currentUser ->
+                userStateService.onUserUpdated(currentUser)
                 storePushNotificationsConfig(
-                    updatedCurrentUser.id,
-                    updatedCurrentUser.name,
+                    currentUser.id,
+                    currentUser.name,
                     userStateService.state !is UserState.UserSet,
                 )
             }
         }
         logger.logI("Initialised: ${buildSdkTrackingHeaders()}")
     }
 
+    private fun ChatEvent.extractCurrentUser(): User? {
+        return when (this) {
+            is HasOwnUser -> me
+            is UserEvent -> getCurrentUser()
+                ?.takeIf { it.id == user.id }
+                ?.mergePartially(user)
+            else -> null
+        }
+    }
+
```

Comment:
The primary fix is here.
We should not replace `currentUser` with the one coming from events with `user` property and not `me`.

Reply:
Could we add this as a comment on the function?